libr/include/r_bin.h

```diff
@@ -170,6 +170,7 @@ typedef struct r_bin_addr_t {
 	ut64 vaddr;
 	ut64 paddr;
 	ut64 haddr;
+	ut64 hvaddr;
```

Comment:
what does it represent? is this really necessary?

Reply:
> what does it represent? is this really necessary?

`haddr` (old code naming) is the physical address of the entry element in the array ... `hvaddr` is the corresponding virtual address ... it's there because IO maps are not available during [`IS_MODE_SET` code execution](https://github.com/kazarmy/radare2/blob/2f17896f4170c308e5cee533e04f669cc4bcccad/libr/core/cbin.c#L1112).

> I think you could use the `rva` function ...

If `rva` stands for relative virtual address, that's not what I want.

Reply:
`rva` is a function which converts selects the right address based on whether `va` is enabled or not. If it is enabled, it also "relocate" the virtual address based on the Base address specified in radare (e.g. with the `-B` option)

I think you don't need IO to get the "virtual" address in this case (e.g. there's `entry->vaddr`). Could you test whether `rva` would do the work or if `entry->vaddr` is enough?

Reply:
Now using `rva` in db40aebaf564ec203d014c1bb33d24aa2acd529f.

Reply:
db40aebaf564ec203d014c1bb33d24aa2acd529f -> cbba72451786356cabd86e192e6038a36fc27a66 since I managed to screw up a `git rebase` and had to force-push.